includes/data-port/class-sensei-data-port-job

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * File containing the Sensei_Data_Port_Job class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Superclass of data import/export jobs. It provides basic functionality like logging, maintaining state, cleanup and
+ * running the data port for each level.
+ */
+abstract class Sensei_Data_Port_Job implements Sensei_Background_Job_Interface {
+	private const OPTION_PREFIX = 'sensei-tools-';
+
+	/**
+	 * An array which holds the results of the data port job and populated in subclasses.
+	 *
+	 * @var array
+	 */
+	protected $results;
+
+	/**
+	 * Unique id for the job.
+	 *
+	 * @var string
+	 */
+	private $job_id;
+
+	/**
+	 * An array containing log entries (e.g. errors). It has the following format:
+	 * [
```

Comment:
There is a WordPress doc standard-ish way to represent this (I think there is an example in the hook docs).